### ISSUE TITLE
[UI] Cluster Templates - Updating a cluster with a template revision having a different CNI should not be allowed

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#21778]
Version: master-head (7/25/19)
Related to #21756     

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1. Create a cluster template/revision T1/R1 having network plugin Canal
2. Create a cluster using this template revision R1
3. Create another revision R2 under template T1 with network plugin Calico
4. Edit the cluster and change the template revision to R2. 

**Result:**
Cluster update DOES provision correctly -- what we are concerned about, is that in the UI we block allowing changing cluster network plugin via edit. So, should we do the same with cluster templates? I would think yes.
We need a way to prevent this through Cluster templates/UI (it is not quite clear how we want to handle this so initially assigned to Prachi)

**Other details that may be helpful:**
Related to rancher/rancher#21756

**Comment by davidnuzik** [comment]
Example of what UI does shown below, does not let us change network plugin:
![image](https://user-images.githubusercontent.com/45179589/61906171-a030bf00-aedf-11e9-88fc-42e27ad01687.png)

Again, not sure if this is something to be address in UI or backend or not. Needs review to make this determination. If for some reason we do NOT want to block changing network plugin on cluster edit, then we need to discuss and ensure this is okay. Then finally, we should also create an issue to have UI remove this so we can change this when not using cluster templates as well.

**Comment by westlywright** [comment]
Cluster templates will mimic the cluster launch page. If you're cloning for a new revision we won't allow you change the network provider to keep it consistent.

**Comment by davidnuzik** [comment]
Version: master-head (8/7/19)

I can confirm that now when I create revisions of a CT I am no longer allowed to change the CNI
![image](https://user-images.githubusercontent.com/45179589/62661989-720f9e00-b927-11e9-989f-a1f014610688.png)

I tested by having my CT have an override and when I edit the cluster I too can't change it from there.

I did notice that I am allowed to change cluster templates though when editing a cluster. This let's me effectively change my CNI. I created another issue to track this here: https://github.com/rancher/rancher/issues/22043
